tests/unit_tests/test_get_metadata.py

```diff
@@ -58,6 +58,20 @@ def test_bcl2fastq_version(runfolder_info):
     assert bcl2fastq_version == "2.20.0.422"
 
 
+def test_get_software_version(runfolder_info):
+    software_versions = runfolder_info.get_software_version(
+        "test_data/210510_M03910_0104_000000000-JHGJL"
+    )
+
+    assert software_versions == {
+        "bcl2fastq": "2.20.0.422",
+        "python": "3.12.0",
+        "yaml": "6.0.1",
+        "Nextflow": "23.04.3",
```

Comment:
I realized another thing: We should make sure it is clear that these are nf-core/Demultiplex versions. We might use other versions of Nextflow and python in the seqreports pipeline. Alternatively, only use keys: `bcl2fastq` and `nf-core/demultiplex`.

Reply:
Also fixed, all versions related to demultiplexing are now gathered in a "Demultiplexing" subsection